apps/emqx/src/emqx_router_helper.erl

```diff
@@ -148,11 +148,13 @@ handle_info({mnesia_table_event, Event}, State) ->
 handle_info({nodedown, Node}, State = #{nodes := Nodes}) ->
     case mria_rlog:role() of
         core ->
+            % TODO
+            % Node may flap, do we need to wait for any pending cleanups in `init/1`
+            % on the flapping node?
+            % This also implies changing lock id to `{?LOCK, Node}`.
```

Comment:
> `{?LOCK, self()}`

It's actually 2 things at once ([doc](https://www.erlang.org/doc/man/global#type-id)):
```
id() = {ResourceId :: [term()](https://www.erlang.org/doc/man/erlang#type-term), LockRequesterId :: [term()]
```

So I'm probably being wrong in that comment I left: `self()` should be fine here, or at least good enough, because there's 32 bits to disambiguate one node incarnation from another:
https://www.erlang.org/doc/apps/erts/erl_ext_dist#new_pid_ext